core/src/proxy/block_engine_stage.rs

```diff
@@ -272,25 +327,39 @@ impl BlockEngineStage {
         ),
         bundle_tx: &Sender<Vec<PacketBundle>>,
         packet_tx: &Sender<PacketBatch>,
-        trust_packets: bool,
+        block_engine_config: &BlockEngineConfig,
         verified_packet_tx: &Sender<(Vec<PacketBatch>, Option<SigverifyTracerPacketStats>)>,
         exit: &Arc<AtomicBool>,
         block_builder_fee_info: &Arc<Mutex<BlockBuilderFeeInfo>>,
+        mut auth_client: AuthServiceClient<Channel>,
+        access_token: Arc<Mutex<Token>>,
+        refresh_token: &mut Token,
+        keypair: Arc<Keypair>,
+        cluster_info: &Arc<ClusterInfo>,
+        connection_timeout: &Duration,
     ) -> crate::proxy::Result<()> {
         const METRICS_TICK: Duration = Duration::from_secs(1);
         const MAINTENANCE_TICK: Duration = Duration::from_secs(10 * 60);
-
+        // Lookahead by Maintenance Tick plus 25%
+        const AUTH_REFRESH_LOOKAHEAD: u64 = MAINTENANCE_TICK
```

Comment:
lookahead kinda confusing, would make more sense if REFRESH_WITHIN_S or something like that saying it'll refresh if the tokens expire within N seconds